js_modules/dagit/packages/ui/src/components/ErrorBoundary.tsx

```diff
@@ -0,0 +1,86 @@
+import React from 'react';
+import styled from 'styled-components/macro';
+
+import {Box} from './Box';
+import {Colors} from './Colors';
+import {Body, Subheading} from './Text';
+import {FontFamily} from './styles';
+
+export type ErrorCollectionContextValue = {
+  errorStackIncluded: boolean;
+  errorCollectionMessage: string;
+  onReportError: (error: Error, context: Record<string, any>) => void;
+};
+
+export const ErrorCollectionContext = React.createContext<ErrorCollectionContextValue>({
+  errorStackIncluded: true,
+  errorCollectionMessage:
+    `Please report this error to the Dagster team via GitHub or Slack. ` +
+    `Refresh the page to try again.`,
+  onReportError: (error, context) => {
+    console.error(error, context);
+  },
+});
+
+interface ErrorBoundaryProps {
+  region: string;
+  resetErrorOnChange?: any[];
+}
+
+interface ErrorBoundaryState {
+  error: Error | null;
+  errorResetPropsValue: string | null;
+}
+
+export class ErrorBoundary extends React.Component<ErrorBoundaryProps, ErrorBoundaryState> {
+  state: ErrorBoundaryState = {error: null, errorResetPropsValue: null};
+
+  static contextType = ErrorCollectionContext;
+
+  componentDidUpdate() {
+    if (
+      this.state.error &&
+      this.state.errorResetPropsValue !== JSON.stringify(this.props.resetErrorOnChange)
+    ) {
+      this.setState({error: null, errorResetPropsValue: null});
+    }
+  }
+
+  componentDidCatch(error: Error, info: any) {
+    // You can also log the error to an error reporting service
+    // Note: Sentry may be blocked by an ad-blocker, can't assume it's available
```

Comment:
Copy-pasted from somewhere?

Reply:
from here https://nextjs.org/docs/advanced-features/error-handling

Reply:
Ahh yes I will rm this before it merges - good catch